### PR TITLE
add static options for some libraries

### PR DESCRIPTION
**Pull request by sarich**
add shared variant for following packages to allow static-only builds:
metis
parmetis
matio
zlib
bzip2

### CODE REVIEW

File: var/spack/repos/builtin/packages/metis/package.py [hunk 1]
```diff
@@ -186,6 +186,13 @@ def install(self, spec, prefix):
 
         if '+shared' in spec:
             options.append('-DSHARED:BOOL=ON')
+        else:
+            rpath_options = []
```
Comment (davydden) [inline var/spack/repos/builtin/packages/metis/package.py]: here and in `parmetis` could you please add a brief in-code comment on what is happening and why you need ot remove those RPATHs?

File: var/spack/repos/builtin/packages/zlib/package.py [hunk 2]
```diff
@@ -40,7 +40,15 @@ class Zlib(AutotoolsPackage):
...
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
+    variant('shared', default=True,
+            description='Enables the build of shared libraries.')
 
     def setup_environment(self, spack_env, run_env):
         if '+pic' in self.spec:
             spack_env.set('CFLAGS', self.compiler.pic_flag)
+
+    def configure(self, spec, prefix):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/zlib/package.py]: that is not the best way to do it, it's an `AutotoolsPackage`, you can add extra (non-prefix) arguments in 
```
def configure_args(self):
  config_args =[]
   if '+shared' not in self.spec:
       config_args.append('--static')
   return config_args
```

**Comment by tgamblin** [comment]
@sarich: can you tweak this per @davydden's request, so we can get it merged?

**Comment by tgamblin** [comment]
@davydden

**Pull request merged**